Add a Thousands of Fish demo (#1340)
godotengine/godot-demo-projects: 3d/thousands_of_fish/README.md

#Thousands of Fish

A simple exmaple of how to animate a mesh using a shader, then create

> thousands of varied copies using using MultiMesh or GPU particles.

Reviewer: ```suggestion
thousands of varied copies using MultiMesh or GPU particles.
```

using MultiMesh or GPU particles.

This is for the ["Animating thousands of fish"](https://docs.godotengine.org/en/stable/tutorials/performance/vertex_animation/animating_thousands_of_fish.html) and ["Controlling thousands of fish with Particles"](https://docs.godotengine.org/en/stable/tutorials/performance/vertex_animation/controlling_thousands_of_fish.html) tutorials in the documentation.

## How it works

The original mesh's surface materials must be replaced or overridden with a ShaderMaterial
that can apply a custom animation shader. A ShaderMaterial has custom data unique to it
and a shader file that can be shared across materials. Replacing a material will remove custom data,
like the colors on the mesh. AnimatedFish demonstrates how to copy properties from the original
material into the new ShaderMaterial.

An alternative method is to right click a mesh -> make unique, then right click the
mesh surface material and choose "Convert to ShaderMaterial". A converted material will make
an editable shader that applies the original properties by default, and can be modified or read
to see how it's done. Each surface should share the same shader to make the animation easier,

> but can still have individual material values if setup manually.

Reviewer: ```suggestion
but can still have individual material values if set up manually.
```
"Setup" is a noun

MultiMesh creates instances of a mesh using per-instance transforms (position, rotation, scale),
but must update transforms on the CPU. Instead, GPUParticles3D updates happen on the GPU,
efficiently moving objects. Each node can function independently, except for this example having

> the model setup in the AnimatedFish mesh node.

Reviewer: ```suggestion
the model set up in the AnimatedFish mesh node.
```
